tests/main/dbus-activation-session-legacy/task.yaml

```diff
@@ -0,0 +1,52 @@
+summary: D-Bus session services support activation without a user of instance systemd
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    snap set system experimental.dbus-activation=true
+
+restore: |
+    systemctl stop private-session-bus.service || true
+    rm -f dbus-launch.env run-session-bus.sh
+    snap unset system experimental.dbus-activation
+    snap unset system experimental.user-daemons
+
+execute: |
+    # This filters out Ubuntu 14.04 and Ubuntu Core respectively
+    if [ "$(command -v systemd-run)" == "" ] || [ "$(command -v dbus-launch)" == "" ]; then
+        exit 0
+    fi
+
+    cat << EOF > run-session-bus.sh
+    #!/bin/sh
+    cd $(pwd)
+    exec dbus-launch --sh-syntax > dbus-launch.env
+    EOF
+    chmod a+x run-session-bus.sh
+
+    echo "Run a private D-Bus session bus as a systemd service"
```

Comment:
CentOS 7 is the other major distro in our battery of tests that requires this.

People could also trigger this case if they start a private session bus separate from the systemd managed one.  This could be to isolate some tests, or because they're using some desktop environment that hasn't updated its startup scripts to take into account the new way of doing things.

It seemed easy enough to run the test on all the systems that supported it, rather than just the ancient ones.